Framework/Core/include/Framework/ASoA.h

```diff
@@ -1024,31 +1024,37 @@ concept can_bind = requires(T&& t) {
   { t.B::mColumnIterator };
 };
 
+template <typename... C>
+concept WithIndexColumns = (is_indexing_column<C> || ...);
+
 template <typename D, typename O, typename IP, typename... C>
 struct TableIterator : IP, C... {
  public:
   using self_t = TableIterator<D, O, IP, C...>;
   using policy_t = IP;
   using all_columns = framework::pack<C...>;
   using persistent_columns_t = framework::selected_pack<soa::is_persistent_column_t, C...>;
-  using indexing_columns_t = framework::selected_pack<is_indexing_t, C...>;
-  constexpr inline static bool has_index_v = framework::pack_size(indexing_columns_t{}) > 0;
   using external_index_columns_t = framework::selected_pack<soa::is_external_index_t, C...>;
   using internal_index_columns_t = framework::selected_pack<soa::is_self_index_t, C...>;
   using bindings_pack_t = decltype([]<typename... Cs>(framework::pack<Cs...>) -> framework::pack<typename Cs::binding_t...> {}(external_index_columns_t{})); // decltype(extractBindings(external_index_columns_t{}));
 
   TableIterator(arrow::ChunkedArray* columnData[sizeof...(C)], IP&& policy)
     : IP{policy},
       C(columnData[framework::has_type_at_v<C>(all_columns{})])...
+  {
+    bind();
+  }
+
+  TableIterator(arrow::ChunkedArray* columnData[sizeof...(C)], IP&& policy) requires (WithIndexColumns<C...>)
+    : IP{policy},
+      C(columnData[framework::has_type_at_v<C>(all_columns{})])...
   {
     bind();
     // In case we have an index column might need to constrain the actual
     // number of rows in the view to the range provided by the index.
     // FIXME: we should really understand what happens to an index when we
     // have a RowViewFiltered.
-    if constexpr (has_index_v) {
-      this->limitRange(this->rangeStart(), this->rangeEnd());
-    }
+    this->limitRange(this->rangeStart(), this->rangeEnd());
```

Comment:
`limitRange` is inherited from an index column, this will probably not compile.

Reply:
As discussed this is actually correct. There is now a requires statement protecting the case.